### PR TITLE
fd_netlink - Handling multipart netlink responses

### PR DESCRIPTION
**Pull request by nbridge-jump**

### CODE REVIEW

File: src/waltz/ip/fd_netlink.c [hunk 1]
```diff
@@ -588,136 +588,146 @@ fd_nl_load_arp_table( fd_nl_t *           nl,
...
+	/* error occurred - don't switch to new routing table */
 
+	return FD_IP_ERROR; /* return failure */
       }
 
-      if( entry->flags & FD_NL_ARP_FLAGS_UNSUPPORTED ) {
-        break;
+      /* we're done if:
+       *        type is NLMSG_DONE
+       *    OR  nlmsg_flags does not contain NLMSG_DONE */
```
Comment (mmcgee-jump) [inline src/waltz/ip/fd_netlink.c:648]:
```suggestion
       *    OR  nlmsg_flags does not contain NLM_F_MULTI */
```

**Comment by mmcgee-jump** [comment]
Can you squash

File: src/waltz/ip/fd_netlink.c [hunk 2]
```diff
@@ -588,136 +588,146 @@ fd_nl_load_arp_table( fd_nl_t *           nl,
...
+
+	rat = RTA_NEXT( rat, ratmsglen );
       }
 
-      rat = RTA_NEXT( rat, ratmsglen );
-    }
+      /* at present, each entry must have at least the following:
+	 FD_NL_ARP_FLAGS_IP_ADDR
+	 FD_NL_ARP_FLAGS_MAC_ADDR
+	 FD_NL_ARP_FLAGS_IFINDEX
```
Comment (ptaffet-jump) [inline src/waltz/ip/fd_netlink.c:720]: Is this supposed to be enforced?

File: src/waltz/ip/fd_netlink.c [hunk 3]
```diff
@@ -588,136 +588,146 @@ fd_nl_load_arp_table( fd_nl_t *           nl,
...
 
+	return FD_IP_ERROR; /* return failure */
       }
 
-      if( entry->flags & FD_NL_ARP_FLAGS_UNSUPPORTED ) {
-        break;
+      /* we're done if:
+       *        type is NLMSG_DONE
+       *    OR  nlmsg_flags does not contain NLM_F_MULTI */
+      done |= ( h->nlmsg_type == NLMSG_DONE ) | !( h->nlmsg_flags & NLM_F_MULTI );
```
Comment (ptaffet-jump) [inline src/waltz/ip/fd_netlink.c:649]: If `h->nlmsg_type == NLMSG_DONE`, should we break? I think the while loop below might have 0 iterations, but we'll still insert an entry.

Comment (nbridge-jump) [reply src/waltz/ip/fd_netlink.c:649]: The flags NLMSG_DONE is set on a valid message, so we need to finish iterating thru the attributes.

**Review by ptaffet-jump — APPROVED** [review]

**Pull request merged**